### ISSUE TITLE
Add blocklist for items on homescreen. 

### ISSUE THREAD

**Issue by MatthewTighe** [mozilla-mobile/fenix#23069]
https://mozilla-hub.atlassian.net/browse/FNXV2-18720

Users can remove items from home screen (tabs, bookmarks, history item)

If an item is removed from one section of the home screen, it may still show up on the others

When an item is removed just the specific page be removed. Other pages from the same website may still show up

Remove option can be access in the context menu by long-pressing on the item

Context menu matches current design for Top Sites and Recent Searches

String “Remove”

Remove items are blocked from showing up on the homescreen, but not from their original location. For example, removing a tab will mean it no longer shows up on Firefox home, but the tab is still open in the tabs tray

If the user visits the same site again in the future, it’s still blocked from showing on home, but not history

Out of Scope

This ticket does not include the option to unblock a tab/site

This ticket doesn’t include groups functionality and shouldn’t change the current behaviour for Recent Searches

**Comment by LaurentiuApahideanSV** [comment]
I tested the issue on Nightly 2022-02-24. When a site is removed from the Recent Bookmarks section of Recently visited section is is removed from all the Home sections (Recent Bookmarks section, Recently visited and Jump back in) if present, furthermore the site will not appear in those sections anymore. The Top sites and search groups are unaffected when a site is removed.

**Comment by highroller** [comment]
There should be an option to unblock a site.  Is there one?

**Comment by highroller** [comment]
How to unblock a website after I blocked it?

**Comment by MatthewTighe** [comment]
@highroller We don't currently offer a way to do that through the app. Clearing the app's cache or reinstalling it should delete the blocklist, but this will erase all other custom settings as well.

**Comment by highroller** [comment]
> @highroller We don't currently offer a way to do that through the app. Clearing the app's data or reinstalling it should delete the blocklist, but this will erase all other custom settings as well.

Is it possible to do it in desktop?  I really don't want to erase all custom settings. Thanks.

**Comment by MatthewTighe** [comment]
Not currently, the blocklist is local to each device and isn't shared. Adding the functionality as a feature would require UX involvement and need to go through our usual prioritization process. I'd suggest filing a feature request issue if it's something you'd like to see added.
